### ISSUE TITLE
[bug]ingress switch routes error

### ISSUE THREAD

**Issue by chenyanter** [Kong/kubernetes-ingress-controller#342]
### Summary

ingress switch routes error

**Kong Ingress controller version**
0.5.0

### Kubernetes version

Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:46:06Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.1", GitCommit:"4ed3216f3ec431b140b1d899130a69fc671678f4", GitTreeState:"clean", BuildDate:"2018-10-05T16:36:14Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}

### What happened
when i create a ingress with two rule, kong create two routes

### Expected behvaior
routes should be only one have two host

### Steps To Reproduce

1.create ingress
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  labels:
    ingress: edge-gateway
    name: edge-gateway
  name: all-to-edge
  namespace: test01
spec:
  rules:
  - host: aa-test01.xxxx.cn
    http:
      paths:
      - backend:
          serviceName: edge-gateway
          servicePort: 80
  - host: bb-test01.xxxx.cn
    http:
      paths:
      - backend:
          serviceName: edge-gateway
          servicePort: 80

2.I saw two routes in kong  which respectively correspond aa-test01.xxxx.cn、bb-test01.xxxx.cn

**Comment by hbagdi** [comment]
Hello @chenyanter 

Thank you for opening this issue but this is by design and not a bug.

It is possible in an Ingress rule that the backend service for each rule is different.
Assuming that each rule will point to the same backend is incorrect.

**Comment by chenyanter** [comment]
The current scenario is to use multiple routes generated by kong
then there were problems accessing the domain name
The upstream address from kong's log is a random IP
The service corresponding to this IP is not  what really want
if this is not a bug, I think I should get the right response

**Comment by chenyanter** [comment]
Or to put it this way, I want to know how to configure ingress so that there can be multiple host in the hosts of the single routes in kong

**Comment by hbagdi** [comment]
> then there were problems accessing the domain name
The upstream address from kong's log is a random IP
The service corresponding to this IP is not what really want
if this is not a bug, I think I should get the right response

I'm not sure if I understand you. Could you rephrase or provide some logs and examples to explain the problem you are facing?

>  I want to know how to configure ingress so that there can be multiple host in the hosts of the single routes in kong

You can specify a wildcard hostname in the hosts field to do routing for multiple subdomains. Combining multiple hosts into a single Route is not possible. I'd like to understand the use-case behind this need. Could you explain a little more?
Thank you!

**Comment by chenyanter** [comment]
1.after recreate ingress the program resolve, so i can't find what happen，i think it maybe a setting error 
2.we can not use wildcard hostname in our scene，but it will resolve after use kong-api-gateway replace our gateway
I'm not good at English, thank you very much
